src/core/namespace.c

```diff
@@ -680,9 +689,29 @@ static int mount_entry_chase(
          * that applies). The result is stored in "location". */
 
         r = chase_symlinks(path, root_directory, 0, &chased);
-        if (r == -ENOENT && m->ignore) {
-                log_debug_errno(r, "Path %s does not exist, ignoring.", path);
-                return 0;
+        if (r == -ENOENT) {
+                bool apivfs = mount_is_apivfs(m);
+
+                /*
+                 * If this is a default API filesystem mount /proc, /sys or
+                 * /dev and if the path does not exist and is not a dangling
+                 * symbolic link, lets not ignore and return success. We will
+                 * create their directories later anyway.
+                 *
+                 * The path at this point is already prefixed with the root
+                 * directory. Also make sure to check the root_directory here,
+                 * this avoids creating directories relative to host's
+                 * root.
+                 */
+                if (apivfs && root_directory && is_symlink(path) == -ENOENT) {
```

Comment:
Oups, should switch to chase_symlinks() with `CHASE_NONEXISTING` flag and update location only when we return positive.